src/base/bittorrent/torrentimpl.cpp

```diff
@@ -96,6 +100,102 @@ namespace
             entryList.emplace_back(setValue.toStdString());
         return entryList;
     }
+
+    lt::announce_entry makeNativeAnnouncerEntry(const QString &url, const int tier)
+    {
+        lt::announce_entry entry {url.toStdString()};
+        entry.tier = tier;
+        return entry;
+    }
+
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+    TrackerEntry fromNativeAnnouncerEntry(const lt::announce_entry &nativeEntry, const lt::info_hash_t &hashes)
+#else
+    TrackerEntry fromNativeAnnouncerEntry(const lt::announce_entry &nativeEntry)
+#endif
+    {
+        TrackerEntry trackerEntry {QString::fromStdString(nativeEntry.url), nativeEntry.tier};
+
+        for (const lt::announce_endpoint &endpoint : nativeEntry.endpoints)
+        {
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+            for (const lt::protocol_version &protocolVersion : {lt::protocol_version::V1, lt::protocol_version::V2})
+            {
+                if (hashes.has(protocolVersion))
+                {
+                    const lt::announce_infohash &infoHash = endpoint.info_hashes[protocolVersion];
+
+                    TrackerEntry::Endpoint trackerEndpoint;
+                    trackerEndpoint.protocolVersion = (protocolVersion == lt::protocol_version::V1) ? 1 : 2;
+                    trackerEndpoint.numSeeds = infoHash.scrape_complete;
+                    trackerEndpoint.numLeeches = infoHash.scrape_incomplete;
+                    trackerEndpoint.numDownloaded = infoHash.scrape_downloaded;
+                    if (infoHash.fails > 0)
+                        trackerEndpoint.status = TrackerEntry::NotWorking;
```

Comment:
Or maybe it should have Updating status regardless of failures if it's currently updating?

Reply:
@Chocobo1, is my suggestion above also incorrect?

Reply:
>@Chocobo1, is my suggestion above also incorrect?

It is hard to say, let's invite testers and see how they perceive.